### PR TITLE
Fixed OwlViTModel inplace operations

### LINKED ISSUE #24525: Backward pass error during OWL-ViT finetuning
### System Info

Google Colab

### Who can help?

@ArthurZucker 
@amyeroberts

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

https://colab.research.google.com/drive/1GfspAHLGTMmzfNShAVebzM04s32q1LKb?usp=sharing

### Expected behavior

I was trying to finetune Owl-ViT, and I came across this backward pass error. I investigated and found out that the error is the operator "/=" which does an inplace operation. 
At lines:

- https://github.com/huggingface/transformers/blob/4e8929dcbb9040f54f52d898a600260f338ef54f/src/transformers/models/owlvit/modeling_owlvit.py#L1296
- https://github.com/huggingface/transformers/blob/4e8929dcbb9040f54f52d898a600260f338ef54f/src/transformers/models/owlvit/modeling_owlvit.py#L1297

### PR DESCRIPTION
**Pull request by pasqualedem**
# What does this PR do?

This PR replaces the "/=" operator in OwlViTModel which causes an error in the backward pass with the non-inplace version.

Fixes #24525 

#24525 

## Who can review?

@ydshieh

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thank you for opening the PR to fix it.
LGTM, but ping a core maintainer @sgugger to have a look too.

File: src/transformers/models/owlvit/modeling_owlvit.py [hunk 1]
```diff
@@ -1293,8 +1293,8 @@ def forward(
             return (pred_logits, image_class_embeds)
 
         # Normalize image and text features
-        image_class_embeds /= torch.linalg.norm(image_class_embeds, dim=-1, keepdim=True) + 1e-6
-        query_embeds /= torch.linalg.norm(query_embeds, dim=-1, keepdim=True) + 1e-6
+        image_class_embeds = image_class_embeds / torch.linalg.norm(image_class_embeds, dim=-1, keepdim=True) + 1e-6
```
Comment (ydshieh) [inline src/transformers/models/owlvit/modeling_owlvit.py:1296]: wait. I think we should do

(torch.linalg.norm(image_class_embeds, dim=-1, keepdim=True) + 1e-6) instead

Comment (ydshieh) [reply src/transformers/models/owlvit/modeling_owlvit.py:1296]:
```suggestion
        image_class_embeds = image_class_embeds / (torch.linalg.norm(image_class_embeds, dim=-1, keepdim=True) + 1e-6)
```

File: src/transformers/models/owlvit/modeling_owlvit.py [hunk 2]
```diff
@@ -1293,8 +1293,8 @@ def forward(
...
+        query_embeds = query_embeds / torch.linalg.norm(query_embeds, dim=-1, keepdim=True) + 1e-6
```
Comment (ydshieh) [inline src/transformers/models/owlvit/modeling_owlvit.py:1297]:
```suggestion
        query_embeds = query_embeds / (torch.linalg.norm(query_embeds, dim=-1, keepdim=True) + 1e-6)
```

**Review by ydshieh — CHANGES_REQUESTED** [review]

**Review by sgugger — APPROVED** [review]
LGTM with @ydshieh suggestions, thanks!

**Comment by ydshieh** [comment]
@pasqualedem Could you apply the commit suggestions, then we can merge 🙏 Thanks.

**Review by ydshieh — APPROVED** [review]
Thanks again!
